crates/bevy_ecs/src/world/error.rs

```diff
@@ -1,12 +1,62 @@
-//! Contains error types returned by bevy's schedule.
+//! Contains error types returned by methods on [`World`].
+//!
+//! [`World`]: crate::world::World
+
+use std::any::TypeId;
 
 use thiserror::Error;
 
-use crate::schedule::InternedScheduleLabel;
+use crate::{component::ComponentId, prelude::*, schedule::InternedScheduleLabel};
 
 /// The error type returned by [`World::try_run_schedule`] if the provided schedule does not exist.
 ///
 /// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
 #[derive(Error, Debug)]
 #[error("The schedule with the label {0:?} was not found.")]
 pub struct TryRunScheduleError(pub InternedScheduleLabel);
+
+/// The error type returned by [`World::get_reflect`] and [`World::get_reflect_mut`].
+///
+/// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
+#[cfg(feature = "bevy_reflect")]
+#[derive(Error, Debug)]
+pub enum GetComponentReflectError {
+    /// There is no [`ComponentId`] corresponding to the given [`TypeId`].
+    ///
+    /// Did you call [`App::register_type()`]?
+    ///
+    /// [`App::register_type()`]: ../../../bevy_app/struct.App.html#method.register_type
+    #[error(
+        "No `ComponentId` corresponding to `TypeId` {0:?} found (did you call App::register_type()?)"
+    )]
+    NoCorrespondingComponentId(TypeId),
+
+    /// The given [`Entity`] does not have a [`Component`] corresponding to the given [`TypeId`].
+    #[error("The given `Entity` {entity:?} does not have a `{component_name:?}` component (`ComponentId` {component_id:?}, which corresponds to `TypeId` {type_id:?})")]
```

Comment:
Nit: I _think_ `ComponentId` might be the same, so it can be removed as well

Reply:
Yup, removed it as well :+1: